VideoLocker/src/main/java/org/edx/mobile/http/NewVersionBroadcastInterceptor.java

```diff
@@ -0,0 +1,75 @@
+package org.edx.mobile.http;
+
+import org.apache.commons.lang3.time.DateFormatUtils;
+import org.edx.mobile.BuildConfig;
+import org.edx.mobile.event.NewVersionAvailableEvent;
+import org.edx.mobile.third_party.versioning.ArtifactVersion;
+import org.edx.mobile.third_party.versioning.DefaultArtifactVersion;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Date;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * An OkHttp interceptor that checks for information about app
+ * updates in the response headers, and broadcasts them on the event
+ * bus if found.
+ */
+public class NewVersionBroadcastInterceptor implements Interceptor {
+    /**
+     * Header field name for the latest version number of the app
+     * that is available in the app stores.
+     */
+    private static final String HEADER_APP_LATEST_VERSION =
+            "EDX-APP-LATEST-VERSION";
+    /**
+     * Header field name for the last date up to which the API used
+     * by the current version of the app will be supported and
+     * maintained.
+     */
+    private static final String HEADER_APP_VERSION_LAST_SUPPORTED_DATE =
+            "EDX-APP-VERSION-LAST-SUPPORTED-DATE";
+
+    @Override
+    public Response intercept(final Chain chain) throws IOException {
+        final Response response = chain.proceed(chain.request());
+
+        final String appLatestVersionString = response.header(HEADER_APP_LATEST_VERSION);
+        final ArtifactVersion appLatestVersion = appLatestVersionString == null ?
+                null : new DefaultArtifactVersion(appLatestVersionString);
+
+        final String lastSupportedDateString =
+                response.header(HEADER_APP_VERSION_LAST_SUPPORTED_DATE);
+        Date lastSupportedDate = null;
+        if (lastSupportedDateString != null) {
+            /* We're using Apache Commons Lang's FastDateFormat here, because it
+             * has improved support for ISO 8601, including UTC offsets and the
+             * "Z" notation.
+             * TODO: Implement this all over the app instead of the existing
+             * SimpleDateFormat.
```

Comment:
It's sending a `+00:00` UTC offset instead of the `Z` character, which means zero offset. `SimpleDateFormat` can parse these via the `XXX` pattern, but it doesn't support one pattern to parse both of these. The `FastDateFormat` in Apache Commons Lang 3 offers the `ZZ` pattern that does parse both of them, amongst other benefits such as thread-safety and caching. It doesn't provide full support for parsing ISO 8601 compliant date formats though, since that requires an actual parser instead of just a formatter, in order to support optional parts (e.g. fractional seconds).

I guess it would be better to use an actual parsing library for ISO 8601. I didn't find any during my initial web search on this, but searching again now I found [ISO 8601 DateFormat][], which provides a parser for ISO 8601 that also implements the `DateFormat` interface. It also links to [ThreeTen Android Backport][], which backports the java.time.* [package][java.time.* package] from Java 8 (which is a full replacement of the old `Date` and `Calendar` APIs). I suppose we should be using one of these alternatives for parsing ISO 8601 dates. What do you think?

[ISO 8601 DateFormat]: https://github.com/nobuoka/java-date-format-iso8601
[ThreeTen Android Backport]: https://github.com/JakeWharton/ThreeTenABP
[java.time.* package]: https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html